controllers/machine_controller.go

```diff
@@ -348,6 +349,12 @@ func (r *MachineReconciler) drainNode(cluster *clusterv1.Cluster, nodeName strin
 		DryRun: false,
 	}
 
+	if !noderefutil.IsNodeReady(node) {
```

Comment:
shouldn't this be checking the node is unreachable instead?

Reply:
Could you please elaborate? The issue stated about "unready" node. How do we want to check if the node is unreachable?

Reply:
I don't believe there is any way for Cluster API to evaluate a node's reachability other than by looking at the conditions of the node in the workload cluster. Or am I missing something?

@michaelgugino @vincepri would we want to always set `drainer.SkipWaitForDeleteTimeoutSeconds`, or only if certain node conditions are set?

Reply:
This feature comes in handy when the node is unreachable see https://kubernetes.io/docs/concepts/architecture/nodes/.
```In some cases when the node is unreachable, the apiserver is unable to communicate with the kubelet on the node. The decision to delete the pods cannot be communicated to the kubelet until communication with the apiserver is re-established. In the meantime, the pods that are scheduled for deletion may continue to run on the partitioned node.```
https://github.com/kubernetes-sigs/cluster-api/issues/1870#issuecomment-580156406

I believe we can assume this when the nodeReady status [1] is unknown [2].

[1] https://github.com/kubernetes-sigs/cluster-api/blob/master/controllers/noderefutil/https://github.com/kubernetes/api/blob/master/core/v1/types.go#L67-L68
[2] https://github.com/kubernetes/api/blob/master/core/v1/types.go#L2318

Reply:
100% agree with what you wrote, but I'm not clear if this is something we should always set, or only when the ready status is unknown?

Reply:
@enxebre did you see my last question?

Reply:
I think I'd be in favour of setting the flag only when the kubelet status is unknown as we deliberately want to let draining move forward in such scenario. It seems safe to let any other unknown scenario where this might happen be visible by blocking draining for now.

Reply:
I've updated the PR to check if node ready status is "`Unknown`" instead of "not `True`". Please check if this matches what we have discussed.